### PR TITLE
FISH-372 Provide option to disable clustering functionality of Hazelcast on Payara Micro (Payara 6)

### PR DESCRIPTION
**Pull request by Pandrex247**
## Description
Port of https://github.com/payara/Payara/pull/5904

### Blockers
None

### New tests
See main PR.

### Testing Performed
Started two micro instances with `--nocluster` - they didn't cluster, and hazelcast was still started
Started two micro instances without `--nocluster` - they clustered
Started two micro instances with `--nohazelcast` - they didn't cluster, and hazelcast wasn't started.

### Testing Environment
Windows 10, JDK 11.

## Documentation
https://github.com/payara/Payara-Documentation/pull/115

## Notes for Reviewers
None

### CODE REVIEW

**Comment by Pandrex247** [comment]
Jenkins test please

File: appserver/tests/fish372-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Fish-372
+
+FISH-372 is the Jira ID of an enhancement which aims at amending the way that some Payara Micro command line options are working, as follows:
+
+* the `--nocluster` option. This option is an existent one and its current role is to disbale the data grid feature. It should be ammended such that to disable only the distributed data grid feature, while still allowing the use of the local data grid.
```
Comment (breakponchito) [inline appserver/tests/fish372-tests/README.md:5]: typo: disable instead of disbale

File: appserver/tests/fish372-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+During the Sprint #12, the Payara Micro code, starting with its 5.2022.3 release, has been modified such that to implement these requirements. A couple of integration tests have been added to the project payara-tests as well. These integration tests have been added to a new maven module, named `fish372-tests`. This was because some issues, related to the source code organization and its compliancy with the IDE conventions, were preventing their implementation in an existent maven module, like for example `micro-programmatic`, where they would probably belong naturally. These issues are presented in the associated Jira ticket.
+
+The integration tests were subject to different scenarii. The scenario that has been considered in the beggining consisted in starting several instances of Payara Micro, using its API, and deploying to them a WAR, hosting a simple JAX-RS endpoint, performing simple operations with local and distributed caches. These simple operations are meant to be successful or to fail, epending of the combination of the command line options, as described later in this document.
```
Comment (breakponchito) [inline appserver/tests/fish372-tests/README.md:10]: typo:  scenari instead of scenarii

Comment (breakponchito) [reply appserver/tests/fish372-tests/README.md:10]: typo: depending instead of epending

File: appserver/tests/fish372-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+In order to run the integration tests a simple `mvn clean install` command isn't enough. This because of the synchronization issue between the Payara Micro server and the integretio-test phase of maven, as explained above. Hence, a couple of scripts have been provided, such that to deal with that:
+
+* run-tests.sh: runs maven while skipping the execution of the integration tests and the post-integration-test phase. This is meant to build the test WAR, to start the Payara Micro required instances and to leave them running, that's why the post-integration-test phase is skipped. Then the script runs the goal verify while skipping both the pre-integration-test phase, such that to avoid starting the instances again, and the post-integration-test, such that to avois stopping the instances. Finally, the script runs the post-integration-test phase, such that to stop the instances, while skipping the pre-integration-test phase which, if ran, would start again the instances.
```
Comment (breakponchito) [inline appserver/tests/fish372-tests/README.md:42]: typo: avoids instead of avois

File: appserver/tests/fish372-tests/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+* stop-payara-instances.sh: this script gets the Payara Micro instances PIDs and stop the associated processes.
+
+These scripts ar shell scripts but, given that they mostly perform maven commands, it's easy to provide the Windows equivalents. For CI testing, these scripts could be used in freestyle jobs or the associated maven commands should be used in maven jobs.
```
Comment (breakponchito) [inline appserver/tests/fish372-tests/README.md:45]: typo: or instead of ar

File: appserver/tests/fish372-tests/pom.xml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      <groupId>jakarta.platform</groupId>
+      <artifactId>jakarta.jakartaee-api</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>jakarta.ws.rs</groupId>
+      <artifactId>jakarta.ws.rs-api</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <!--dependency>
```
Comment (breakponchito) [inline appserver/tests/fish372-tests/pom.xml:70]: this commented config, can be removed?

**Review by breakponchito — APPROVED** [review]
LGTM

**Comment by breakponchito** [comment]
I identified some typos on the README file, please check

**Comment by Pandrex247** [comment]
Typos will be resolved in a separate PR

**Pull request merged**